core/os/os2/process.odin

```diff
@@ -166,15 +166,15 @@ Process_Info :: struct {
 
 	This procedure obtains an information, specified by `selection` parameter of
 	a process given by `pid`.
-	
-	Use `free_process_info` to free the memory allocated by this procedure. In
-	case the function returns an error it may only have been an error for one part
-	of the information and you would still need to call it to free the other parts.
+
+	Use `free_process_info` to free the memory allocated by this procedure. The
+	`free_process_info` procedure needs to be called, even if this procedure
+	returned an error, as some of the fields have been allocated.
```

Comment:
I think this should be "the fields may have been allocated."

I also think the docs should say something about how even if an error is returned, the `.fields` can be checked for things that have been successfully retrieved, and that the error is for the other fields that couldn't be retrieved.

It does get pretty lengthy but it is a weird procedure.

Reply:
I am contemplating whether it was a good choice for the API, but still not entirely sure that's the best solution. Thinking about windows, you get process information from different places, depending on which information you want to retrieve.

For example, if you want to know the command line of the process, on windows you'd read the other process' PEB. If you want to know parent pid of a process, the only way I found without the NTDLL weirdness is to make a toolhelp32 process snapshot. And on linux all of these things are done by reading from different files of the `/proc` filesystem (`/proc/cmdline`, `/proc/stat`, etc).

With this API I didn't want it to be too slow by doing every single thing a user "might" want to have and filling out the process info struct completely. So I added a mechanism that would allow the user to "select" what info they want, so that the inner bodies of the `process_info` procedures can be optimized to only retrieve the things the user has requested.

...

Got distracted, lost my train of thought.

So if the procedure, instead of how it works currently, just retrieved all the information, that would be a waste of resources, i.e. "why is this reading process memory and failing when I just asked for parent process id", and if it was made in separate calls, it would lead to a question of "why is the process being opened and closed 8 times for each process im asking".

So I decided on an API that would allow the user to chose what they want and what they don't want, to provide the best speed and cleanliness for whatever set of information you may want to retrieve.

This was in response to "this is a weird procedure", because I do think it is... unique, and I'm wondering if I'm overthinking things. It's not exactly the type of thing you do millions times per second, so perhaps a better way to think about it is in terms of how easy it would be to handle errors instead? In whichcase maybe separating API out into separate procedures is a better idea. What are your thoughts?

Reply:
IMO the API is pretty nice as is. I like the idea of requesting fields and getting whatever was available back, instead of an error because one of the things wasn't available. Splitting it up comes with it's own downsides as you also wrote.

Thinking of a `top` like program, you would probably request every field and display whatever was available. Other uses could request whatever they really need and take the returned error as a "fact" and bubble it up.

In terms of handling errors, the only information you might lose in this API is if multiple fields were erroneous in different ways, just one of the errors would be returned back, IMO that is acceptable. The general flow I see would be: request x, y, z; get x back (in the .fields bit_set); see the error for why y and z weren't retrieved. 

Some functionality is just a bit unique and requires more explaining / documentation.

Reply:
I won't split it up, at least not now. I think this PR is ready for merging